src/remap/function/parse_grok.rs

```diff
@@ -0,0 +1,162 @@
+use remap::prelude::*;
+use std::collections::BTreeMap;
+use std::sync::Arc;
+
+#[derive(Clone, Copy, Debug)]
+pub struct ParseGrok;
+
+impl Function for ParseGrok {
+    fn identifier(&self) -> &'static str {
+        "parse_grok"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                accepts: |v| matches!(v, Value::Bytes(_)),
+                required: true,
+            },
+            Parameter {
+                keyword: "pattern",
+                accepts: |v| matches!(v, Value::Bytes(_)),
+                required: true,
+            },
+        ]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Result<Box<dyn Expression>> {
+        let value = arguments.required_expr("value")?;
+
+        let patternbytes = arguments
+            .required_literal("pattern")?
+            .as_value()
+            .clone()
+            .try_bytes()?;
+
+        let patternstr = String::from_utf8_lossy(&patternbytes).into_owned();
+
+        let mut grok = grok::Grok::with_patterns();
+        let pattern = Arc::new(grok.compile(&patternstr, true).map_err(|e| e.to_string())?);
```

Comment:
It's probably a good idea to add a test to see how this error is represented. I don't think we've had any other functions that return any error other than `required_*` errors, so I'm not sure what the final error message will look like.